File: src/main/java/com/sica/backend/entity/Employee.java
package com.sica.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Entity
@Table(name = "employees")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Employee {

    @Id
    private String personId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "person_id")
    private Person person;

    @Column(nullable = false, unique = true)
    private String employeeNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "area_id", nullable = false)
    private Area area;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Shift shift;

    @Column(nullable = false)
    private LocalTime entryTime;

    @Column(nullable = false)
    private LocalTime exitTime;
}

File: src/main/kotlin/com/sica/backend/entity/Employee.kt
package com.sica.backend.entity

import jakarta.persistence.*
import java.time.LocalTime

@Entity
@Table(name = "employees")
data class Employee(
    @Id
    val personId: String,

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "person_id")
    val person: Person,

    @Column(nullable = false, unique = true)
    val employeeNumber: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "area_id", nullable = false)
    val area: Area,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val shift: Shift,

    @Column(nullable = false)
    val entryTime: LocalTime,

    @Column(nullable = false)
    val exitTime: LocalTime
)
